server/tests/test_log_sanitizer.py

```
@@ -0,0 +1,190 @@
+        old = os.environ.pop("FLASK_SECRET_KEY", None)
+        try:
+            assert hash_for_log("user123") == "?"
+        finally:
+            if old is not None:
+                os.environ["FLASK_SECRET_KEY"] = old
+            _get_log_hash_salt.cache_clear()
+
+    def test_with_secret_key_returns_hex(self):
+        from utils.log_sanitizer import _get_log_hash_salt
+        _get_log_hash_salt.cache_clear()
+        os.environ["FLASK_SECRET_KEY"] = "test-secret-key"
```

Ruff (0.15.11)

[error] Possible hardcoded password assigned to: "FLASK_SECRET_KEY"

(S105)